src/fa/maps_/preview.py

```
@@ -67,7 +71,17 @@ def create_large_preview(
     scmap = get_scmap_file(mapdir)
     assert scmap is not None
-    mapdata = map_data_from_scmap(scmap).scale_image_by(scale)
+    try:
+        mapdata = map_data_from_scmap(scmap).scale_image_by(scale)
+    except struct.error as e:
+        logger.warning("Could not extract map data from .scmap file: %s", e)
+        dds_file = get_dds_file(mapdir)
+        assert dds_file is not None
+        image = image_from_dds(dds_file)
+        mapdata = MapData(image.width(), image.height(), image).scale_image_by(scale)
+    except Exception as e:
```

Ruff (0.15.7)

[warning] Do not catch blind exception: `Exception`

(BLE001)